src/Okta.Sdk/Internal/AbstractResourceTypeResolver.cs

```diff
@@ -5,21 +5,34 @@
 
 using System;
 using System.Collections.Generic;
-using System.Text;
 
 namespace Okta.Sdk.Internal
 {
     /// <summary>
     /// Resolves a resource type based on the resource data. Base class for all resource type resolvers.
     /// </summary>
     /// <typeparam name="T">The base type of resource to resolve.</typeparam>
-    public abstract class AbstractResourceTypeResolver<T>
+    public abstract class AbstractResourceTypeResolver<T> : IResourceTypeResolver
     {
         /// <summary>
         /// Get the resolved resource type given its <paramref name="data"/>.
         /// </summary>
         /// <param name="data">The resource data.</param>
         /// <returns>The resource type.</returns>
-        public abstract Type GetResolvedType(IDictionary<string, object> data);
+        public Type GetResolvedType(IDictionary<string, object> data)
+        {
+            var resourceType = GetResolvedTypeInternal(data);
+
+            if (ResourceTypeResolverFactory.GetResolver(resourceType) == null)
+            {
+                return resourceType;
+            }
+
+            var resolverType = ResourceTypeResolverFactory.GetResolver(resourceType);
```

Comment:
Add comment:

// If there is a more specific resolver available, resolve again recursively